hyperspy/tests/test_events.py

```diff
@@ -0,0 +1,378 @@
+def f_c(*args): pass
+def f_d(a, b, c): pass
+
+
+class TestEventsSignatures(EventsBase):
+
+    def setUp(self):
+        self.events = he.Events()
+        self.events.a = he.Event()
+
+    def test_basic_triggers(self):
+        self.events.a.connect(lambda *args, **kwargs: 0)
+        self.events.a.connect(lambda: 0, None)
+        self.events.a.connect(lambda x: 0, 1)
+        self.events.a.connect(lambda x, y: 0, 2)
+        self.events.a.connect(lambda x, y=988:
+                              nt.assert_equal(y, 988), 1)
+        self.events.a.connect(lambda x, y=988:
+                              nt.assert_not_equal(y, 988), 2)
+        self.events.a.trigger(2, 5)
+
+        nt.assert_raises(ValueError, self.events.a.trigger)
+        nt.assert_raises(ValueError, self.events.a.trigger, 2)
+        self.events.a.trigger(2, 5, 8)
+
+    def test_connected(self):
+        self.events.a.connect(f_a)
+        self.events.a.connect(f_a, None)
+        self.events.a.connect(f_b, 2)
+        self.events.a.connect(f_c, 5)
+        self.events.a.connect(f_d, 'auto')
+
+        ref = {'all': set([f_a]), 0: set([f_a]), 1: set(), 2: set([f_b]),
+               3: set([f_d]), 5: set([f_c]),
+               None: set([f_a, f_b, f_c, f_d])}
+        for k, v in ref.iteritems():
+            con = self.events.a.connected(k)
+            nt.assert_equal(con, v)
+
+        con = self.events.a.connected()
+        nt.assert_equal(con, ref[None])
+
+    @nt.raises(TypeError)
+    def test_type(self):
+        self.events.a.connect('f_a')
+
+
+class TestTriggerArgResolution(EventsBase):
+
+    def setup(self):
+        self.events = he.Events()
+        self.events.a = he.Event(arguments=['A', 'B'])
+        self.events.b = he.Event(arguments=['A', 'B', 'C'])
+        self.events.c = he.Event()
+
+    def test_nargs_resolution(self):
+        self.events.a.connect(lambda x=None: nt.assert_equal(x, None), 0)
+        self.events.a.connect(lambda x: nt.assert_equal(x, 'vA'), 1)
+        self.events.a.connect(lambda x, y:
+                              nt.assert_equal((x, y), ('vA', 'vB')), 2)
+        self.events.a.connect(lambda x, A:
+                              nt.assert_equal((x, A), ('vA', 'vB')), 2)
+        self.events.a.connect(lambda x, y=None, A=None:
+                              nt.assert_equal((x, y, A),
+                                              ('vA', 'vB', None)), 2)
+        self.events.a.connect(lambda B, A:
+                              nt.assert_equal((B, A), ('vA', 'vB')), 2)
+        self.events.a.connect(lambda B, y:
+                              nt.assert_equal((B, y), ('vA', 'vB')), 2)
+        self.events.a.connect(lambda B, A=None:
+                              nt.assert_equal(B, 'vA'), 1)
+        self.events.a.connect(lambda B, y=None:
+                              nt.assert_equal(B, 'vA'), 1)
+        # Update if internal code changes, not testing this line:
+        self.events.c._connected = self.events.a._connected
```

Comment:
I'm pretty confident this was left behind by accident. Just remove it unless it somehow breaks a test.